### PR TITLE
Desktop deployment

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?
Introducing a flag to install automate in desktop mode. To install automate with Desktop mode enable run `chef-automate deploy --product desktop`. To enable Desktop mode in the hab studio, run `enable_desktop` in the hab studio. To see that desktop is enabled in the UI, look for the "https://localhost/automate.conf.js" request on any page in Automate. Ensure the response has "desktop" in the products list. 

### :chains: Related Resources

https://github.com/chef/automate/issues/3244

### :athletic_shoe: How to Build and Test the Change
1. `build components/automate-cli && build components/automate-deployment && start_all_services`
1. Run this command for reference below. `curl -f --insecure -H "api-token: ${token}" https://localhost/automate.conf.js`
1. Run `enable_desktop`
1. Run this command to simulate a Desktop install. The below response with "desktop" added should be returned`curl -f --insecure -H "api-token: ${token}" https://localhost/automate.conf.js`
```
parseStaticAutomateConfig({
  "products": [
    "desktop"
  ]
})
```
### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/deploy.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ var deployCmdFlags = struct {
 	enableChefServer                bool
 	enableDeploymentOrderStressMode bool
 	enableWorkflow                  bool
+	enableDesktop                   bool
```
Comment (jaym) [inline components/automate-cli/cmd/chef-automate/deploy.go:53]: we'd like to avoid having the enable flags. that was the old way

Comment (vjeffrey) [reply components/automate-cli/cmd/chef-automate/deploy.go:53]: what's new the new way, @jaym ?

Comment (lancewf) [reply components/automate-cli/cmd/chef-automate/deploy.go:53]: @vjeffrey, Jay and I talked and the new way is a lot easier. With Jay's help, I think we have a solution for all our needs for desktop. 

The only question now is would the below line be acceptable to install automate with desktop? 
`chef-automate deploy --product desktop`

Comment (vjeffrey) [reply components/automate-cli/cmd/chef-automate/deploy.go:53]: i think so! i'll send this to jessica to make sure! 🎉

File: dev/config.toml [hunk 2]
```diff
@@ -83,6 +83,7 @@ shared_buffers = "1GB"
     deployment_type = "local"
     manifest_cache_expiry = "0s"
     package_cleanup_mode = "disabled"
+    products = ["automate", "desktop"]
```
Comment (lancewf) [inline dev/config.toml:86]: This is temporary and will not be merged in.

File: components/automate-cli/cmd/chef-automate/deploy.go (hunk 1 above)
Comment (vjeffrey) [reply components/automate-cli/cmd/chef-automate/deploy.go]: confirmed this flag is 👍 with product

File: components/automate-deployment/pkg/services/services.go [hunk 3]
```diff
@@ -102,13 +102,6 @@ func ValidateProductDeployment(products []string) error {
 		}
 	}
 
-	for requiredCollectionName, requiredCollection := range requiredCollectionSet {
```
Comment (lancewf) [inline components/automate-deployment/pkg/services/services.go:105]: This prevented dependencies of type "product" from the products.meta file from automatically being install. Before the command to install desktop would have been `chef-automate deploy --product automate --product desktop` with this removed it is `chef-automate deploy --product desktop`. 
Desktop is defined below
```
    {
      "name": "desktop",
      "type": "product",
      "dependencies": ["automate"],
      "services": []
    }
```
With this code removed the dependence "automate" is automatically installed without explicitly declaring to install it.

File: components/automate-deployment/pkg/services/services_test.go [hunk 4]
```diff
@@ -236,9 +236,6 @@ func TestValidateProductDeployment(t *testing.T) {
 	t.Run("error on base collection", func(t *testing.T) {
 		assert.Error(t, ValidateProductDeployment([]string{"automate", "core"}))
 	})
-	t.Run("error unspecified product dependencies", func(t *testing.T) {
-		assert.Error(t, ValidateProductDeployment([]string{"workflow"}))
-	})
```
Comment (lancewf) [inline components/automate-deployment/pkg/services/services_test.go:241]: This is a test for the constraint that was removed above.

**Pull request merged**